### PR TITLE
Move test examples on end of test run them in different CI job 

### PR DESCRIPTION
**Pull request by Czaki**
# Description

As test examples do not use `make_napari_viewer` to create a viewer it may happen that not every widget is cleaned after these test executions. It may cause the following test will fail during setup phase.  

https://github.com/napari/napari/runs/6796938348?check_suite_focus=true#step:7:275

To avoid this false-positive test failure this PR moves all test examples to the end. 

Zrzut ekranu z 2022-06-09 01-37-43

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# References

Few observed failures. For example https://github.com/napari/napari/runs/6796938348?check_suite_focus=true#step:7:275

# How has this been tested?

- [x] using `pytest --co .` to verify test order. 

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Comment by Carreau** [comment]
I think a better fix would/should be to have a cleanup fixture for test example that make sure everything is cleaned up, but that's already an improvement.

**Review by Carreau — APPROVED** [review]

**Comment by Carreau** [comment]
I restarted failing test to make sure this was not actually due to this change.

**Comment by tlambert03** [comment]
maybe we should run example tests in their own subtest, outside of the main tests?  They're the most problematic tests, and area really just there to catch an API breakage, right?  We could run one python/backend version on linux?

**Comment by Czaki** [comment]
> maybe we should run example tests in their own subtest, outside of the main tests? They're the most problematic tests, and area really just there to catch an API breakage, right? We could run one python/backend version on linux?

I like this idea. especially connected with information from this comment https://github.com/napari/napari/pull/4646#issuecomment-1150785513

**Comment by tlambert03** [comment]
great, wanna do that here?  maybe

```
  test_examples:
    name: test examples
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-python@v3
        with:
          python-version: 3.9
      - uses: tlambert03/setup-qt-libs@v1
      - name: Install this commit
        run: |
          pip install --upgrade pip
          pip install .[all,testing]

      - name: Test
        uses: GabrielBB/xvfb-action@v1
        with:
          run: python -m pytest --color=yes napari/_tests/test_examples.py

```

and then make sure they're being ignored otherwise somehow?

**Comment by Czaki** [comment]
> and then make sure they're being ignored otherwise somehow?

so maybe we should move it outside test tree? Or do you prefer to use some pytest marking?

**Comment by tlambert03** [comment]
> so maybe we should move it outside test tree? Or do you prefer to use some pytest marking?

that occurred to me too.  whatever you think makes the most sense

**Comment by Czaki** [comment]
@tlambert03 did you know this test fail?

**Comment by tlambert03** [comment]
hmm... I recall it being mysterious once before:  https://github.com/napari/napari/pull/4059

File: napari/conftest.py [hunk 1]
```diff
@@ -503,23 +518,23 @@ def pytest_generate_tests(metafunc):
...
+        os.path.join("napari", "utils"),
+        os.path.join("napari", "layers"),
+        os.path.join("napari", "components"),
+        os.path.join("napari", "settings"),
+        os.path.join("napari", "plugins"),
+        os.path.join("napari", "_vispy"),
+        os.path.join("napari", "_qt"),
+        os.path.join("napari", "qt"),
+        os.path.join("napari", "_tests"),
+        os.path.join("napari", "_tests", "test_examples.py"),
```
Comment (Czaki) [inline napari/conftest.py:530]: Windows paths

File: napari/view_layers.py [hunk 2]
```diff
@@ -109,7 +109,7 @@ def _merge_layer_viewer_sigs_docs(func):
 
 
 _viewer_params = inspect.signature(Viewer).parameters
-_dims_params = inspect.signature(Dims).parameters
+_dims_params = Dims.__fields__
```
Comment (Czaki) [inline napari/view_layers.py:112]: I do not like this but:
```ipython
In [6]: inspect.signature(Dims)
Out[6]: <Signature (**kwargs)>
```

**Review by goanpeca — APPROVED** [review]

File: napari/view_layers.py (hunk 2 above)
Comment (tlambert03) [reply napari/view_layers.py:112]: works for me

File: .github/workflows/test_pull_requests.yml [hunk 3]
```diff
@@ -194,3 +194,22 @@ jobs:
...
+      - uses: tlambert03/setup-qt-libs@v1
+      - name: Install this commit
+        run: |
+          pip install --upgrade pip
+          pip install setuptools tox tox-gh-actions
+
+      - name: Test
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          run: tox -e py39-linux-pyside2-examples
```
Comment (tlambert03) [inline .github/workflows/test_pull_requests.yml]:
```suggestion
          run: tox -e py39-linux-pyside2-examples

```

File: .github/workflows/test_comprehensive.yml [hunk 4]
```diff
@@ -150,3 +150,22 @@ jobs:
...
+      - uses: tlambert03/setup-qt-libs@v1
+      - name: Install this commit
+        run: |
+          pip install --upgrade pip
+          pip install setuptools tox tox-gh-actions
+
+      - name: Test
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          run: tox -e py39-linux-pyside-examples
```
Comment (tlambert03) [inline .github/workflows/test_comprehensive.yml]:
```suggestion
          run: tox -e py39-linux-pyside2-examples
```

?

File: tox.ini [hunk 5]
```diff
@@ -83,13 +83,19 @@ commands =
...
     # do not add ignores to this line just to make headless tests pass.
     # nothing outside of _qt or _vispy should require Qt or make_napari_viewer
     headless: python -m pytest --color=yes --basetemp={envtmpdir} --ignore napari/_vispy \
         --ignore napari/_qt --ignore napari/_tests --ignore tools \
         --json-report --json-report-file={toxinidir}/report-{envname}.json {posargs}
 
 
+[testenv:py{38,39,310}-{linux,macos,windows}-{pyqt5,pyside2}-examples]
+commands =
+    python -m pytest napari/_tests/test_examples.py --color=yes --basetemp={envtmpdir} {posargs}
```
Comment (tlambert03) [inline tox.ini]:
```suggestion
    python -m pytest napari/_tests/test_examples.py -v --color=yes --basetemp={envtmpdir} {posargs}
```

(few enough that it's nice to see)

**Review by tlambert03 — APPROVED** [review]
looks good to me!  couple very minor suggestions

File: napari/view_layers.py (hunk 2 above)
Comment (Czaki) [reply napari/view_layers.py:112]: I would like to know what happens that is examples are executed then the signature is properly recognized.

**Pull request merged**